src/main/java/com/jenkinsci/plugins/badge/action/BadgeSummaryAction.java

```diff
@@ -23,101 +23,83 @@
  */
 package com.jenkinsci.plugins.badge.action;
 
-import com.jenkinsci.plugins.badge.BadgePlugin;
-import hudson.markup.RawHtmlMarkupFormatter;
-import java.io.IOException;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import org.apache.commons.lang.StringEscapeUtils;
+import org.apache.commons.lang.StringUtils;
 import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
-import org.kohsuke.stapler.export.Exported;
-import org.kohsuke.stapler.export.ExportedBean;
 
-@ExportedBean(defaultVisibility = 2)
-public class BadgeSummaryAction extends AbstractAction {
+/**
+ * Common action for build summaries.
+ */
+public class BadgeSummaryAction extends AbstractBadgeAction {
 
     private static final long serialVersionUID = 1L;
-    private static final Logger LOGGER = Logger.getLogger(BadgeSummaryAction.class.getName());
-
-    private final String iconPath;
-    private String summaryText = "";
 
-    public BadgeSummaryAction(String iconPath) {
-        this.iconPath = iconPath;
-    }
-
-    /* Action methods */
-    public String getUrlName() {
-        return "";
+    public BadgeSummaryAction(String id, String icon, String text, String cssClass, String style, String link) {
```

Comment:
This is an incompatible change. It breaks https://github.com/jenkinsci/groovy-postbuild-plugin

Reply:
please crate new issue for that. This change was merged now, so we need to fix it in new PR.